compiler/rustc_typeck/src/expr_use_visitor.rs

```diff
@@ -265,6 +265,9 @@ impl<'a, 'tcx> ExprUseVisitor<'a, 'tcx> {
                                     if def.variants.len() > 1 {
                                         needs_to_be_read = true;
                                     }
+                                } else {
+                                    // If it is not ty::Adt, then it is a MultiVariant
```

Comment:
I feel like MultiVariant is confusing here, maybe say something like interger/array etc which can take various different values.

Reply:
I think there is still and edge case here:

```rust
fn invalid() { // ICE on edition 2021
    let mut arr = [1, 3, 4, 5];
    
    let mref = &mut arr;
    
    let c = || match arr { // arr will be captured when it doesn't need to
    //  this throws a borrow check error on edition 2018 and I think it will be the same case here.
        [_, _, _, _] => println!("A")
    };
    
    println!("{:#?}", mref);
}



fn valid() { // this doesn't throw borrow check error so `arr` should captured when we put match in closure
    let mut arr = [1, 3, 4, 5];
    
    let mref = &mut arr;
    
   // outside closure
    match arr {
        [_, _, _, _] => println!("A")
    };
    
    println!("{:#?}", mref);
}
```

Reply:
Actually it's probably related to #87987. Because this is a Slice Pat and would be handled not here 😅

Reply:
Yes, if you think of any other related edge cases let me know